kmultiversx/src/kmultiversx/kdist/mx-semantics/mandos.md

```diff
@@ -102,6 +113,21 @@ Only take the next step once both the Elrond node and Wasm are done executing.
             #fi
         [simplification]
 
+    syntax MapBytesToBytes  ::= #removeReservedKeys ( MapBytesToBytes ) [function]
+ // ----------------------------------------------------------------------------------------
+    rule #removeReservedKeys(.MapBytesToBytes)
+        => .MapBytesToBytes
+    rule #removeReservedKeys(wrap(Key) Bytes2Bytes|-> Value M)
+        =>  #if #hasPrefix(Bytes2String(Key), "ELROND")
+            #then #removeReservedKeys(M)
```

Comment:
this does not preserve definedness: if wrap(Key) is in M, then, on this branch, the LHS is not defined, while the RHS is. The same applies below.

Reply:
Should I use `keys_list(M)` and iterate over the list?

Reply:
That would work. Another option would be to add `requires notBool wrap(Key) in M`